### PR TITLE
Update tables and use of aggregates

### PR DESCRIPTION
**Pull request by Graycemuthui**
### Description
 In this project I achieved the following:
- Added an extra column to the previous table.
- Inserted additional rows to the table.
- Updated, deleted, committed and also rolled back data inputted.
- Created queries for the data.
- Used proper Gitflow.

### Screenshots.
1. Updated the animals table by setting the species column to digimon for all animals that have a name ending in mon.

DIGIMON

2 .Update the animals table by setting the species column to pokemon for all animals that don't have species already set.

pokemon

3. Delete and rollback data.

transcation 1

4. Delete all animals born after Jan 1st, 2022.

delete

5. Update all animals' weight to be their weight multiplied by -1.

weight to -1

6. Update all animals' weights that are negative to be their weight multiplied by -1.

weight multiply -1

7. How many animals are there?

query 1

8. How many animals have never tried to escape?

query 2

9. What is the average weight of animals?

query 3

10. Who escapes the most, neutered or not neutered animals?

query 5

11. What is the minimum and maximum weight of each type of animal?

query 6

12. What is the average number of escape attempts per animal type of those born between 1990 and 2000?

query 7

### Changes made.
- UPDATE the animals table by setting the species column to unspecified, then roll back the change

rollback

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,10 +1,59 @@
...
+
+-- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
```
Comment (dasileker) [inline queries.sql:34]: - [ ] Nice job on updating the database but please  you Inside a transaction:
* `UPDATE` the animals table by setting the species column to unspecified, then roll back the change

File: queries.sql [hunk 2]
```diff
@@ -1,10 +1,59 @@
...
+-- Delete all animals born after Jan 1st, 2022.
+DELETE FROM animals where date_of_birth > '2022-01-01'
+
+-- Update all animals' weight to be their weight multiplied by -1.
+UPDATE animals SET weight_kg  = (weight_kg * (-1));
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
```
Comment (dasileker) [inline queries.sql:46]: - [ ] Also add How many animals have never tried to escape

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Graycemuthui,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**

- PR is very good ✅.
- The readme  is well done ✅.

## Required Changes ♻️
_Check the comments under the review._

- Please make sure to add all the transactions into your `queries.sql` right now you are missing a few.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (Graycemuthui) [reply queries.sql:46]: Hello this is number 8 in the pull request and it is also in the vs code.

File: queries.sql (hunk 1 above)
Comment (Graycemuthui) [reply queries.sql:34]: Okay let me work on this.

File: queries.sql (hunk 2 above)
Comment (dasileker) [reply queries.sql:46]: They are different, null is not unspecified

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @Graycemuthui 

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**